src/serac/physics/utilities/finite_element_state.hpp

```diff
@@ -163,17 +173,92 @@ class FiniteElementState {
   {
     static_assert(std::is_constructible_v<Tensor, mfem::ParFiniteElementSpace*>,
                   "Tensor must be constructible with a ptr to ParFESpace");
-    return std::make_unique<Tensor>(&space_);
+    return std::make_unique<Tensor>(&retrieve(space_));
   }
 
 private:
+  /**
+   * @brief A helper type for uniform semantics over owning/non-owning pointers
+   */
+  template <typename T>
+  using MaybeOwner = std::variant<T*, std::unique_ptr<T>>;
+
+  /**
+   * @brief Retrieves a reference to the underlying object in a MaybeOwner
+   * @param[in] obj The object to dereference
+   */
+  template <typename T>
+  static T& retrieve(MaybeOwner<T>& obj)
```

Comment:
Is it possible to overload the `*` operator for this?

Reply:
To my knowledge it's not possible to implement nonmember operator overloads for nested template classes (the compiler just can't figure out the lookups, I think).